stdlib/public/core/ClosedRange.swift

```diff
@@ -47,26 +47,64 @@ public struct ClosedRangeIndex<Bound> : Comparable
     }
   }
   
-  internal func _predecessor(upperBound limit: Bound) -> ClosedRangeIndex {
+  internal func _predecessor(
+    lowerBound: Bound, upperBound limit: Bound
```

Comment:
At this point, now that you are passing both bounds, I think it would make sense to just move this code into its only callsite.